Meta/.shell_include.sh

```diff
@@ -3,7 +3,10 @@
 # SC2034: "Variable appears unused. Verify it or export it."
 #         Those are intentional here, as the file is meant to be included elsewhere.
 
-SUDO="sudo"
+# NOTE: If using another privillege escalation binary make sure it is configured or has the appropiate flag
+# 	to keep the current environment variables in the launched process (in sudo's case this is achieved
+# 	through the -E flag described in SUDO(8).
```

Comment:
The appropriate manpage would be `sudo(8)` (not `SUDO(8)`), wouldn't it?

Reply:
> Also, this does not appear to entirely fix #15894, as it does not introduce support for `doas`.

Wouldn't the user just need to replace `SUDO="sudo -E"` with `SUDO="doas"`?

Reply:
No, doas only preserves environment variables called out in a config file. We would need to dump the env variables we need to a config file for `doas` users to separately add to their env.

Reply:
Indeed, but until our scripts do that I wouldn't consider the issue fixed. Otherwise one could argue that they just need to replace `${SUDO} -E` with a properly configured `doas` right now, and we don't need this patch at all.

Reply:
> No, doas only preserves environment variables called out in a config file. We would need to dump the env variables we need to a config file for doas users to separately add to their env.

I believe when using the keepenv option in doas it should preserve all variables:
`keepenv: Environment variables other than those listed in [doas(1)](https://man.openbsd.org/doas.1) are retained when creating the environment for the new process.`

Or am I misunderstanding this?😅

Reply:
> Indeed, but until our scripts do that I wouldn't consider the issue fixed. Otherwise one could argue that they just need to replace `${SUDO} -E` with a properly configured `doas` right now, and we don't need this patch at all.

That's fair :),  would this mean that this pr can be merged without closing/referencing the original issue or should we not bother until we find a good UX way to do this?

Reply:
> I believe when using the keepenv option in doas it should preserve all variables:
keepenv: Environment variables other than those listed in [doas(1)](https://man.openbsd.org/doas.1) are retained when creating the environment for the new process.

The linked manpage does not mention a `keepenv` option, so that appears to be nonstandard? Or are you planning to ship a custom configuration file along with it?

> That's fair :), would this mean that this pr can be merged without closing/referencing the original issue or should we not bother until we find a good UX way to do this?

At least with the current state, we should drop the "Fixes <number>" from the PR description and commit message. Whether it's fine to merge the PR anyways the maintainers will have to decide, because right now the change is pretty much a no-op.

Reply:
> The linked manpage does not mention a keepenv option, so that appears to be nonstandard? Or are you planning to ship a custom configuration file along with it?

My bad, I didnt include the doas.conf(5) manpage link (https://man.openbsd.org/doas.conf)

> At least with the current state, we should drop the "Fixes " from the PR description and commit message. 

Will do

> Whether it's fine to merge the PR anyways the maintainers will have to decide, because right now the change is pretty much a no-op.

Pretty much yeah, it does however make it easier to change privilege escalation binary by just changing a single var instead of going through the scripts and removing all (sudo/pls)-specific usages of the -E flag.

Reply:
Per the examples on https://man.openbsd.org/doas.conf.5#EXAMPLES

There's a `setenv` option that says "preserve this env variable when running this command", and a `keepenv` command that's a bigger hammer that does indeed say "keep all of it"

Reply:
> Per the examples on https://man.openbsd.org/doas.conf.5#EXAMPLES
> 
> There's a `setenv` option that says "preserve this env variable when running this command", and a `keepenv` command that's a bigger hammer that does indeed say "keep all of it"

So when using doas users would have to enable the keepenv options right? Is this something we should allude to in the NOTE? Or do you think that saying that the privilege escalation binary the use has to keep the environment variables enough?